File: packages/protocol/src/task-events.ts
import { z, taskIdSchema } from "./zod";

export const taskEventKindSchema = z.enum([
  "accepted",
  "progress",
  "need_user_input",
  "completed",
  "failed",
]);

export const taskEventPayloadSchema = z.object({
  taskId: taskIdSchema,
  eventKind: taskEventKindSchema,
  summary: z.string().min(1),
});

export type TaskEventPayload = z.infer<typeof taskEventPayloadSchema>;

File: packages/protocol/src/protocol/task_events.py
from typing import Literal

from pydantic import BaseModel, ConfigDict


class TaskEventPayload(BaseModel):
    model_config = ConfigDict(extra="forbid")

    taskId: str
    eventKind: Literal["accepted", "progress", "need_user_input", "completed", "failed"]
    summary: str
